### PR TITLE
feat: Add diamond tier to fetch-sponsors

### LINKED ISSUE #633: Suggestion: Update fetch sponsors script
### What problem do you want to solve?

The fetch sponsors script doesn't support the diamond tier.

### What do you think is the correct solution?

Add support for the diamond tier.

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What changes did you make? (Give an overview)

Updated the `fetch-sponsors.js` script to identify diamond tier sponsors (> =$5,000).

#### Related Issues

fixes #633

### CODE REVIEW

**Review by harish-sethuraman — APPROVED** [review]
LGTM. waiting for second review.

**Review by amareshsm — APPROVED** [review]
LGTM.

**Pull request merged**
